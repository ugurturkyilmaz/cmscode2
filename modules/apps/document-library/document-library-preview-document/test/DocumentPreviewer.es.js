/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {cleanup, render} from '@testing-library/react';
import React from 'react';

import DocumentPreviewer from '../src/main/resources/META-INF/resources/preview/js/DocumentPreviewer.es';

describe('document-library-preview-document', () => {
	afterEach(cleanup);

	it('renders a document previewer with ten pages and the first page rendered', () => {
		const {asFragment} = render(
			<DocumentPreviewer
				baseImageURL="http://localhost/document-images/"
				initialPage={1}
				spritemap="icons.svg"
				totalPages={10}
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});

	it('renders a document previewer with nineteen pages and the fifth page rendered', () => {
		const {asFragment} = render(
			<DocumentPreviewer
				baseImageURL="http://localhost/document-images/"
				initialPage={5}
				spritemap="icons.svg"
				totalPages={19}
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
