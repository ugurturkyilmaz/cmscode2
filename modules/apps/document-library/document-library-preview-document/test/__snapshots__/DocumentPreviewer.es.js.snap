// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`document-library-preview-document renders a document previewer with nineteen pages and the fifth page rendered 1`] = `
<DocumentFragment>
  <div
    class="preview-file"
  >
    <div
      class="preview-file-container preview-file-max-height"
    >
      <img
        class="preview-file-document preview-file-document-fit"
        src="http://localhost/document-images/?previewFileIndex=5"
      />
    </div>
    <div
      class="preview-toolbar-container"
    >
      <div
        class="floating-bar btn-group"
        role="group"
      >
        <div
          class="btn-group"
          role="group"
        >
          <button
            class="btn-floating-bar btn-floating-bar-text btn btn-primary"
            title="click-to-jump-to-a-page"
            type="button"
          >
            page 5 / 19
          </button>
        </div>
        <button
          class="btn-floating-bar btn btn-monospaced btn-primary"
          title="page-above"
          type="button"
        >
          <svg
            class="lexicon-icon lexicon-icon-caret-top"
            role="presentation"
          >
            <use
              xlink:href="#caret-top"
            />
          </svg>
        </button>
        <button
          class="btn-floating-bar btn btn-monospaced btn-primary"
          title="page-below"
          type="button"
        >
          <svg
            class="lexicon-icon lexicon-icon-caret-bottom"
            role="presentation"
          >
            <use
              xlink:href="#caret-bottom"
            />
          </svg>
        </button>
        <div
          class="separator-floating-bar"
        />
        <button
          class="btn-floating-bar btn btn-monospaced btn-primary"
          title="expand"
          type="button"
        >
          <svg
            class="lexicon-icon lexicon-icon-full-size"
            role="presentation"
          >
            <use
              xlink:href="#full-size"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`document-library-preview-document renders a document previewer with ten pages and the first page rendered 1`] = `
<DocumentFragment>
  <div
    class="preview-file"
  >
    <div
      class="preview-file-container preview-file-max-height"
    >
      <img
        class="preview-file-document preview-file-document-fit"
        src="http://localhost/document-images/?previewFileIndex=1"
      />
    </div>
    <div
      class="preview-toolbar-container"
    >
      <div
        class="floating-bar btn-group"
        role="group"
      >
        <div
          class="btn-group"
          role="group"
        >
          <button
            class="btn-floating-bar btn-floating-bar-text btn btn-primary"
            title="click-to-jump-to-a-page"
            type="button"
          >
            page 1 / 10
          </button>
        </div>
        <button
          class="btn-floating-bar btn btn-monospaced btn-primary"
          disabled=""
          title="page-above"
          type="button"
        >
          <svg
            class="lexicon-icon lexicon-icon-caret-top"
            role="presentation"
          >
            <use
              xlink:href="#caret-top"
            />
          </svg>
        </button>
        <button
          class="btn-floating-bar btn btn-monospaced btn-primary"
          title="page-below"
          type="button"
        >
          <svg
            class="lexicon-icon lexicon-icon-caret-bottom"
            role="presentation"
          >
            <use
              xlink:href="#caret-bottom"
            />
          </svg>
        </button>
        <div
          class="separator-floating-bar"
        />
        <button
          class="btn-floating-bar btn btn-monospaced btn-primary"
          title="expand"
          type="button"
        >
          <svg
            class="lexicon-icon lexicon-icon-full-size"
            role="presentation"
          >
            <use
              xlink:href="#full-size"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
